### PR TITLE
New IA - Authentication & Authorization

### PR DESCRIPTION
**Pull request by DavidRollins**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link
https://deploy-preview-5690--tyk-docs.netlify.app/docs/nightly/api-management/authentication-authorization/
<br>

### Description
1. Describe your changes in detail. Why is this change required? What problem does it solve?
Update the authentication and authorization page to consolidate content around authentication types and how to secure your apis in Tyk

     2. Please explain the change to the reviewer. Pay special attention to changes that are hard to spot
       2.1 Updated the navigation to each section, added badges, edited the intro
       2.2 Reorganized the page into key sections:
image

       2.3 Updated the menu.yaml and deleted the redundant content

     3. @mentions of the person to review the proposed changes. They need to be able to know the topic well in order to approve it.
@letzya 
@sharadregoti 
<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Added a new comprehensive documentation page for authentication and authorization in Tyk, covering various methods such as OAuth 2.0, JWT, mTLS, and more.
- Included detailed setup instructions, examples, and configurations for implementing these authentication methods.
- Updated the navigation menu to include the new "Authentication and Authorization" page, enhancing accessibility to the documentation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>authentication-authorization.md</strong><dd><code>Add detailed authentication and authorization documentation</code></dd></summary>
<hr>

tyk-docs/content/api-management/authentication-authorization.md

<li>Added comprehensive documentation on authentication and authorization <br>methods in Tyk.<br> <li> Detailed setup instructions for OAuth 2.0, JWT, mTLS, and other <br>authentication methods.<br> <li> Included examples and configurations for various authentication flows.<br> <li> Provided guidance on integrating external OAuth providers and using Go <br>plugins.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5690/files#diff-adf85bfca0c3de1ac4a98340e9d875563c52e9036bd1994306dfa7c9daf9d544">+1735/-0</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu to include authentication documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

<li>Added a new menu entry for "Authentication and Authorization".<br> <li> Updated the navigation structure to include the new documentation <br>page.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5690/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by letzya** [comment]
@DavidRollins the PR has conflicts and also I don't see that the deleted pages have been added as aliases

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 1]
```diff
@@ -0,0 +1,1739 @@
...
+**Response:**
+
+The response provides an access token that can be used by the client to access resources.
+
+```yaml
+{
+  "access_token": "abcd1234token",
+  "expires_in": 3600,
+  "token_type": "bearer"
+}
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: In the [original response](https://tyk.io/docs/basic-config-and-security/security/authentication--authorization/oauth2-0/username-password-grant/#response), it contains an extra field called `refresh_token`

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 2]
```diff
@@ -0,0 +1,1739 @@
...
+##### Request an Access Token
+
+The client application authenticates directly with the authorization server using its client credentials (client ID and secret) to request an access token.
+
+```bash
+curl -X POST \
+  https://tyk.cloud.tyk.io/oauth-api/oauth/token/ \
+  -H 'Authorization: Basic ZWQ1OTE1OGZhMjM0NGU5NGIzZTYyNzhlOGFiODUxNDI6TUdRM056RTJNR1F0WVRVeVpDMDBaVFZsTFdKak1USXRNakUyTVRNMU1tRTNOMk0x' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'grant_type=client_credentials&client_id=ed59158fa2344e94b3e6278e8ab85142'
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: Client secret is also passed in the [original doc](https://tyk.io/docs/basic-config-and-security/security/authentication--authorization/oauth2-0/client-credentials-grant/#token-request). Need to verify once which one is correct.

The `data` table below also has to be modified.

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 3]
```diff
@@ -0,0 +1,1739 @@
...
+##### Use the Refresh Token to Request a New Access Token
+
+When the access token expires, the client can use the refresh token to obtain a new access token without requiring user interaction.
+
+```bash
+curl -X POST \
+  https://tyk.cloud.tyk.io/oauth-api/oauth/token/ \
+  -H 'Authorization: Basic ZWQ1OTE1OGZhMjM0NGU5NGIzZTYyNzhlOGFiODUxNDI6TUdRM056RTJNR1F0WVRVeVpDMDBaVFZsTFdKak1USXRNakUyTVRNMU1tRTNOMk0x' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'grant_type=refresh_token&refresh_token=NWQzNGVhMTItMDE4Ny00MDFkLTljOWItNGE4NzI1ZGI1NGU2&client_id=ed59158fa2344e94b3e6278e8ab85142'
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: Same as below, client_secret missing

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 4]
```diff
@@ -0,0 +1,1739 @@
...
+
+##### Allow Developers to Upload Certificates
+
+Instead of manually creating keys, you can allow developers to upload their own certificates via the Developer Portal.
+
+1. **Create a Policy:** Create a policy for the API you set up earlier.
+2. **Create a Catalog Entry:** Create a catalog entry for this policy.
+3. **Request a Key through the Portal:** As a developer, request a key for the API through the Portal. This will present a screen where the developer can upload their public certificate.
+
+   {{< img src="/img/2.10/client_mtls_add_cert.png" alt="Add Key Certificate" >}}
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: It seems we are not using the correct image here. The original has a different [image](https://tyk.io/docs/basic-config-and-security/security/mutual-tls/client-mtls/#developer-portal---self-serve-cert-trust).

In the original docs, this image is used in the previous [step](https://tyk.io/docs/basic-config-and-security/security/mutual-tls/client-mtls/#quickstart).

File: tyk-docs/content/basic-config-and-security/security.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ Tyk supports TLS connections and Mutual TLS. All TLS connections also support HT
 
 ### Trusted Certificates
 
-As part of using Mutual TLS, you can create a list of [trusted certificates]({{< ref "basic-config-and-security/security/mutual-tls/concepts#certificates" >}}).
+As part of using Mutual TLS, you can create a list of [trusted certificates]({{< ref "/api-management/authentication-authorization#concepts#certificates" >}}).
```
Comment (sharadregoti) [inline tyk-docs/content/basic-config-and-security/security.md]: Multiple fragments do not work. Please check and correct everywhere.

File: tyk-docs/content/getting-started/key-concepts/authentication.md [hunk 6]
```diff
@@ -215,7 +215,7 @@ securitySchemes: {
 
 All you need to do in the Tyk configuration is to enable the authentication and specify the header details.
 
-For more configuration options check the [JWT documentation]({{< ref "/content/basic-config-and-security/security/authentication-authorization/json-web-tokens.md" >}}).
+For more configuration options check the [JWT documentation]({{< ref "/api-management/authentication-authorization#use-json-web-tokens-jwt.md" >}}).
```
Comment (sharadregoti) [inline tyk-docs/content/getting-started/key-concepts/authentication.md]: `.md` prefix at the end also does not work. Please check & fix everywhere.

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 7]
```diff
@@ -0,0 +1,2176 @@
...
+  -H 'Authorization: Basic dXNlcm5hbWU6cGFzc3dvcmQ='
+```
+
+**Request:**
+
+| Parameter       | Value                                                           |
+| --------------- | -------------------------------------------------------------- |
+| **Method**      | `GET`                                                            |
+| **URL**         | The API endpoint for the protected resource.                     |
+| **Authorization** | Basic authorization using base64 encoded credentials, e.g., `dXNlcm5hbWU6cGFzc3dvcmQ=`. |
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: For basic authentication, we missed some important section from the [docs](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/basic-auth/#protect-your-api-with-basic-authentication). Is there any rationale behind it?

The same has not been conveyed in the IA as well.

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 8]
```diff
@@ -0,0 +1,2176 @@
...
+
+
+### Integrate External OAuth Middleware
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md:939]: We missed some important section from the [docs](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/ext-oauth-middleware/#introspection). Is there any rationale behind it?

The same has not been conveyed in the IA as well.

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 9]
```diff
@@ -0,0 +1,2176 @@
...
+| **URL**         | The API endpoint for the protected resource. |
+| **Authorization** | Bearer token, e.g., `Bearer VALIDATED_ACCESS_TOKEN`. |
+
+### Authenticate Using Go Plugins
+
+Go Plugin Authentication allows you to implement custom authentication logic using the Go programming language. This method is useful for scenarios where you need to implement specialized authentication mechanisms that are not natively supported by Tyk.
+To learn more about using Tyk Golang Plugins, go [here](/plugins/supported-languages/golang/#authentication-with-a-golang-plugin)
+
+
+### Sign Requests with HMAC
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]: We missed some important section from the [docs](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/hmac-signatures/#a-sample-signature-generation-snippet). Is there any rationale behind it?

The same has not been conveyed in the IA as well.

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 10]
```diff
@@ -0,0 +1,2176 @@
...
+| --------------- | ---------------------------------- |
+| **Method**      | `GET`                                |
+| **URL**         | The API endpoint for the protected resource. |
+| **Authorization** | Bearer token, e.g., `Bearer ID_TOKEN`. |
+
+
+
+## Combine Authentication Methods
+
+### Multiple (Chained) Authentication
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md:1988]: Let's add this [reference](https://tyk.io/docs/basic-config-and-security/security/authentication-authorization/multiple-auth/#enable-multi-chained-authentication-with-tyk-operator), which is missing in ours.

**Review by sharadregoti — CHANGES_REQUESTED** [review]

File: tyk-docs/content/api-management/authentication-authorization.md [hunk 11]
```diff
@@ -0,0 +1,2804 @@
...
+{{< /note >}}
+
+
+[OpenID Connect](https://openid.net/developers/how-connect-works) (OIDC) builds on top of OAuth 2.0, adding authentication. You can secure your APIs on Tyk by integrating with any standards compliant OIDC provider using [JSON Web Tokens]({{< ref "#use-json-web-tokens-jwt" >}}) (JWTs).
+JWTs offer a simple way to use the third-party Identity Provider (IdP) without needing any direct integration between the Tyk and 3rd-party systems.
+
+To integrate a 3rd party OAuth2/OIDC IdP with Tyk, all you will need to do is ensure that your IdP can issue OAuth2 JWT access tokens as opposed to opaque tokens.
+
+The client application authenticates with the IdP which then provides an access token that is accepted by Tyk. Tyk will take care of the rest, ensuring that the rate limits and quotas of the underlying identity of the bearer are maintained across JWT token re-issues, so long as the "sub" (or whichever identity claim you chose to use) is available and consistent throughout and the policy that underpins the security clearance of the token exists too.
+
```
Comment (sharadregoti) [inline tyk-docs/content/api-management/authentication-authorization.md]:
```suggestion
```

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**